connectivity/check/policy.go

```diff
@@ -623,90 +665,52 @@ func (t *Test) CiliumLogs(ctx context.Context) {
 	}
 }
 
-// parseCiliumPolicyYAML decodes policy yaml into a slice of CiliumNetworkPolicies.
-func parseCiliumPolicyYAML(policy string) (cnps []*ciliumv2.CiliumNetworkPolicy, err error) {
-	if policy == "" {
+// ParsePolicyYAML decodes a yaml file into a slice of policies.
+func ParsePolicyYAML[T runtime.Object](input string, scheme *runtime.Scheme) (output []T, err error) {
```

Comment:
Nice approach!

NIT: Why this function is public?

Reply:
I exported the generalized functions to make it possible to reuse them from external hooks as well.

Reply:
Gotcha, thanks!